lib/libutils/ext/arch/arm/mcount_a64.S

```diff
@@ -80,4 +80,6 @@ FUNC __ftrace_return, :
 END_FUNC __ftrace_return
 #endif
 
+BTI(emit_aarch64_feature_1_and     GNU_PROPERTY_AARCH64_FEATURE_1_BTI)
```

Comment:
This needs to be outside the `#if defined(CFG_TA_GPROF_SUPPORT) || defined(CFG_FTRACE_SUPPORT)`. Otherwise when both features are disabled, we get an (almost) empty object file without the BTI feature bit and when building a shared library the whole `libutils.so` is without BTI.

Since it is too easy to make mistakes with BTI, I think we need a post-link phase that checks that BTI is indeed enabled in all user space binaries when `CFG_TA_BTI=y`.

Reply:
I agree. Perhaps doing a readelf with -n option to check if note property is present in the elf or in so. I am still learning the OP-TEE make system and how it functions. So any quick suggestions on  which file these checks should be added would help.

Reply:
Yes, something like that. Fully linked binaries should be relatively easy (TA `.elf`, shared library `.so`), archives are a bit more complex since we want to check that each object files in the archive has the note with the BTI bit, but then there is the special case of `setjmp_a64.o` with Clang... Perhaps a custom Python script based on ELFFile would be better.

I will propose something.

Reply:
Should we consider adding this in a separate PR ?

Reply:
We could. On the other hand this can probably wait a bit since I do not expect major conflicts on rebasing or anything. I will try to do something today, it shouldn't be that hard ;-)

Reply:
Sure and thanks for helping out with this.